src/engine/benchmark/source/hlp/hlp_bench.cpp

```diff
@@ -0,0 +1,384 @@
+{
+    std::string ret, folder_separator;
+    if(unixFormat){
+        folder_separator = '/';
+        ret += getRandomString(len);
+    }
+    else {
+        folder_separator = '\\';
+        ret += "C:\\";
+    }
+    for(int i = 0; i < length; i++){
+        ret += getRandomString(len) + folder_separator ;
+    }
+
+    return ret;
+}
+
+static std::string createRandomDomain(int len, bool withSubdomain = false)
+{
+    std::string ret;
+    ret += "www.";
+    if(withSubdomain){
+        ret += getRandomString(len) + '.';
+    }
+    ret += getRandomString(len);
+    ret += ".com/";
+    ret += getRandomString(len);
+
+    return ret;
+}
+
+static std::string createMap(int len)
+{
+    std::string ret;
+    for(int i = 0; i < len; ++i){
+        ret += "key";
+        ret += std::to_string(i);
+        ret += '=';
+        ret += getRandomString(10);
+        if (i != (len - 1))
+            ret += ' ';
+    }
+    ret += ';';
+
+    return ret;
+}
+
+// TODO Probably need a way to mix-match a variable number of all our parsers
+static const char *logQL_expression = "<source.address> - <JSON> - [<timestamp/APACHE>] \"<http.request.method> "
+    "<url> HTTP/<http.version>\" <http.response.status_code> <http.response.body.bytes> \"-\" "
+    "\"<user_agent.original>\"";
+static const char *event = "monitoring-server - {\"data\":\"this is a json\"} - [29/May/2017:19:02:48 +0000] "
+    "\"GET https://user:password@wazuh.com:8080/status?query=%22a%20query%20with%20a%20space%22#fragment "
+    "HTTP/1.1\" 200 612 \"-\" \"Mozilla/5.0 (Windows NT 6.1; rv:15.0) Gecko/20120716 Firefox/15.0a2\"";
+
+static void capture_expr_parse(benchmark::State &state)
+{
+    auto expr = getRandomCapExpr(state.range(0));
+    for (auto _ : state)
+    {
+        auto parseOp = getParserOp(expr);
+        if(!parseOp){
+            state.SkipWithError("Invalid expr");
+        }
+    }
+}
+BENCHMARK(capture_expr_parse)->Range(8, 8 << 10);
+
+static void capture_var_expr_parse(benchmark::State &state)
+{
+    auto expr = getRandomCapExprVariable(state.range(0));
+    for (auto _ : state)
+    {
+        auto parseOp = getParserOp(expr);
+        if(!parseOp){
+            state.SkipWithError("Invalid expr");
+        }
+    }
+}
+BENCHMARK(capture_var_expr_parse)->Range(8, 8 << 10);
+
+static void match_literal_range(benchmark::State &state)
+{
+    srand((unsigned)time(NULL));
+    std::string ev = getRandomString(state.range(0));
+    for (auto _ : state)
+    {
+        const char *eventIt = ev.c_str();
+        if(!matchLiteral(&eventIt, ev)){
+            state.SkipWithError("Parser failed");
+        }
+    }
+}
+BENCHMARK(match_literal_range)->Range(8, 8 << 11);
+
+static void execute_parsers(benchmark::State &state)
+{
+    auto parseOp = getParserOp(logQL_expression);
+    for (auto _ : state)
+    {
+        auto result = parseOp(event);
+    }
+}
+BENCHMARK(execute_parsers);
+
+static void full_pass(benchmark::State &state)
```

Comment:
`execute_parsers` and  `full_pass` probably need a better name too

Reply:
done